.github/workflows/build.yml

```diff
@@ -31,6 +31,13 @@ jobs:
     - name: Smoke test (Mac, Windows)
       run: npm run test
       if: runner.os != 'Linux'
+    - name: Upload coverage reports to Codecov
+      uses: codecov/codecov-action@def81595a67c83cbb0a42ba9b624b97f4c7deb3d #v3.1.4
```

Comment:
Do we need a very specific version here? Would it work with `uses: codecov/codecov-action`?

Reply:
Yes for security purposes we ping to sha so that even if other sha's have any security vulnerabilities or if this particular commit has security issues we will know easily and this has been followed across AKS atlanta team repo's.

Reply:
@peterbom more about pinning to commit sha [here](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) 

```
Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.

```